src/Lean/Elab/Term.lean

```diff
@@ -1769,7 +1769,7 @@ private def mkConsts (candidates : List (Name × List String)) (explicitLevels :
     return (const, projs) :: result
 
 def resolveName (stx : Syntax) (n : Name) (preresolved : List Syntax.Preresolved) (explicitLevels : List Level) (expectedType? : Option Expr := none) : TermElabM (List (Expr × List String)) := do
-  addCompletionInfo <| CompletionInfo.id stx stx.getId (danglingDot := false) (← getLCtx) expectedType?
+  withRef stx <| addCompletionInfo <| CompletionInfo.id stx stx.getId (danglingDot := false) (← getLCtx) expectedType?
```

Comment:
I think this is the right place to add `withRef`:


We need it for `mkConsts` used here  in `resolveName`, `mkConsts` but and which doesn't take syntax anymore.